packages/flutter/lib/src/widgets/drag_target.dart

```diff
@@ -535,6 +568,11 @@ class DragTarget<T> extends StatefulWidget {
   /// the target.
   final DragTargetLeave onLeave;
 
+  /// Called when a [Draggable] moves within this [DragTarget].
+  ///
+  /// Note that this includes entering and leaving the target.
+  final DragTargetMove onMove;
```

Comment:
Actually, why do you need the detailsBuilder? Doesn't the onMove callback give you enough information to change the state from the outside to what you want?

Reply:
See my original bug description: https://github.com/flutter/flutter/issues/58982
It would be unergonomic to trigger a second rebuild outside (of the initial movement rebuild) when the user wants to update children.

Reply:
Why would there be a second rebuild? If I remember correctly, onMove is executed between frame (before DragTarget can rebuild in response to the move), so any state changes that are triggered in the onMove would be applied in the same frame as the DragTarget rebuild, no?

Reply:
Here's didDrop (existing code) and didMove (new code, modeled on the existing patterns). It calls setState() first, then calls any relevant callbacks. 

```
  void didDrop(_DragAvatar<Object> avatar) {
    assert(_candidateAvatars.contains(avatar));
    if (!mounted)
      return;
    setState(() {
      _candidateAvatars.remove(avatar);
    });
    if (widget.onAccept != null)
      widget.onAccept(avatar.data as T);
    if (widget.onAcceptWithDetails != null)
      widget.onAcceptWithDetails(DragTargetDetails<T>(data: avatar.data as T, offset: avatar._lastOffset));
  }

  void didMove(_DragAvatar<Object> avatar) {
    if (!mounted)
      return;
    if (widget.detailedMovementBuilder != null)
      setState(() {});
    if (widget.onMove != null)
      widget.onMove(DragTargetDetails<dynamic>(data: avatar.data, offset: avatar._lastOffset));
  }
```

Even if it were to work without needing to add it to the next frame, that's not apparent to the caller. I feel that most Dart/Flutter developers will instead assume that callbacks may be called during rebuild, and will always schedule it on the next frame, e.g. would use onMove:
```
onMove:(details) => SchedulerBinding.instance.addPostFrameCallback((_) => setState(()=>_coords = details.offset));
```

Reply:
The order of calling setState and calling the callbacks doesn't matter here. setState just marks this widget to need rebuild in the next frame. It doesn't actually do the rebuild.

Scheduling a postFrameCallback like that in response to a gesture callback is also very unusual. You wouldn't do that in a GestureDetector's onDrag/onTap/etc. either.

Reply:
I've confirmed that the postFrameCallback is not required, i.e. I can just call setState() in the onMove() callback. However, I think many flutter coders instinctivly avoid setState() within a build _method_ out of habit, even though it is ok in this context since it is part of a callback outside of the build phase. But yes, I agree it's not needed. More thoughts in the other comment below.